### PR TITLE
Builds the latest version of srsran

### PR DESCRIPTION
**Pull request by saltiyazan**
- Builds the latest version of srsRAN
- Adds build packages

### CODE REVIEW

File: snap/snapcraft.yaml [hunk 1]
```diff
@@ -45,10 +47,11 @@ parts:
...
-      ./usr/share/srslte/sib.conf.example: ./usr/share/srslte/sib.conf
-      ./usr/share/srslte/user_db.csv.example: ./usr/share/srslte/user_db.csv
+      ./usr/share/srsran/ue.conf.example: ./usr/share/srsran/ue.conf
+      ./usr/share/srsran/drb.conf.example: ./usr/share/srsran/drb.conf
+      ./usr/share/srsran/enb.conf.example: ./usr/share/srsran/enb.conf
+      ./usr/share/srsran/epc.conf.example: ./usr/share/srsran/epc.conf
+      ./usr/share/srsran/rr.conf.example: ./usr/share/srsran/rr.conf
+      ./usr/share/srsran/sib.conf.example: ./usr/share/srsran/sib.conf
+      ./usr/share/srsran/user_db.csv.example: ./usr/share/srsran/user_db.csv
+      ./usr/share/srsran/rb.conf.example: ./usr/share/srsran/rb.conf
```
Comment (sanchezfdezjavier) [inline snap/snapcraft.yaml:57]: Did we intentionally remove `rb.conf.example`?

Comment (saltiyazan) [reply snap/snapcraft.yaml:57]: in the 20_10_1 version you mean? it wasn't present among the example config files.
now in 22_10 it is, and srsenb would crash if it doesn't find it.

Comment (sanchezfdezjavier) [reply snap/snapcraft.yaml:57]: Understood, resolved.

**Review by sanchezfdezjavier — APPROVED** [review]

File: snap/snapcraft.yaml [hunk 2]
```diff
@@ -13,21 +13,23 @@ apps:
...
+  /etc/srsran:
+    bind: $SNAP/usr/share/srsran
 parts: 
-  srslte:
+  srsran:
     plugin: cmake
     cmake-parameters:
       - -DCMAKE_BUILD_TYPE=RelWithDebInfo
       - -DENABLE_ASAN=True
       - -DCMAKE_INSTALL_PREFIX=/usr
```
Comment (gruyaume) [inline snap/snapcraft.yaml:26]: Can we remove this parameter and the `prime` section altogether? Sorry, this is not directly related to this PR.

**Review by ghislainbourgeois — APPROVED** [review]

File: snap/snapcraft.yaml (hunk 2 above)
Comment (gruyaume) [reply snap/snapcraft.yaml:26]: On a similar note, does it work with `core22` instead of `core20`?

File: snap/snapcraft.yaml [hunk 3]
```diff
@@ -13,21 +13,23 @@ apps:
   srsepc:
```
Comment (gruyaume) [inline snap/snapcraft.yaml:13]: (not this line) the top level `version` needs to be updated as well

**Review by gruyaume — CHANGES_REQUESTED** [review]

File: snap/snapcraft.yaml (hunk 3 above)
Comment (saltiyazan) [reply snap/snapcraft.yaml:13]: definitely :)
✅

File: snap/snapcraft.yaml (hunk 2 above)
Comment (saltiyazan) [reply snap/snapcraft.yaml:26]: core22 wouldn't be able to find `libzmq3-dev`.
There would be solutions for that, like building zeromq from source :)

Comment (saltiyazan) [reply snap/snapcraft.yaml:26]: > Can we remove this parameter and the `prime` section altogether? Sorry, this is not directly related to this PR.

This was added to have the example config files in the `usr` directory, but we could test other options to copy those files :)

Comment (gruyaume) [reply snap/snapcraft.yaml:26]: We can keep it as is, it doesn't matter much honestly
